gwtquery-core/src/main/java/com/google/gwt/query/client/GQuery.java

```diff
@@ -280,17 +227,41 @@ public TagWrapper(int wrapDepth, String preWrap, String postWrap) {
         return (GQuery) o;
       }
       if (o instanceof Function) {
-        return $((Function) o);
-      }
-      if (JsUtils.isElement(o)) {
-        return $(JsUtils.<Element> cast(o));
-      }
-      if (o instanceof JavaScriptObject) {
-        return $((JavaScriptObject) o);
+        return $(((Function) o).getElement());
```

Comment:
I'm concerned about this change for the future. It will be easy to screw up this part of code and introduce a loop. Do we have test for each that covers each possibility ?